src/TrigScint/TrigScintQIEDigiProducer.cxx

```diff
@@ -0,0 +1,148 @@
+#include "TrigScint/TrigScintQIEDigiProducer.h"
+#include "Framework/RandomNumberSeedService.h"
+#include "Framework/Exception/Exception.h"
+
+#include <iostream>
+
+namespace ldmx {
+
+  TrigScintQIEDigiProducer::TrigScintQIEDigiProducer
+  (const std::string& name, Process& process) :
+    Producer(name, process) {    
+  }
+
+  TrigScintQIEDigiProducer::~TrigScintQIEDigiProducer() {
+  }
+
+  void TrigScintQIEDigiProducer::configure(Parameters& parameters) {
+
+    // Configure this instance of the producer
+    stripsPerArray_   =
+      parameters.getParameter< int >("number_of_strips");
+    numberOfArrays_   =
+      parameters.getParameter< int >("number_of_arrays");
+    meanNoise_        =
+      parameters.getParameter< double >("mean_noise");
+    mevPerMip_        =
+      parameters.getParameter< double >("mev_per_mip");
+    pePerMip_         =
+      parameters.getParameter< double >("pe_per_mip");
+    inputCollection_  =
+      parameters.getParameter< std::string >("input_collection");
+    inputPassName_    =
+      parameters.getParameter< std::string >("input_pass_name" );
+    outputCollection_ =
+      parameters.getParameter< std::string >("output_collection");
+    verbose_          =
+      parameters.getParameter< bool >("verbose");
+
+    // QIE specific parameters initialization
+    maxts_ 	      	=
+      parameters.getParameter< int >("maxts");
+    toff_overall_     	=
+      parameters.getParameter< double >("toff_overall");
+    input_pulse_shape_ 	=
+      parameters.getParameter< std::string >("input_pulse_shape");
+    tdc_thr 		=
+      parameters.getParameter< double >("tdc_thr");
+    pedestal 		=
+      parameters.getParameter< double >("pedestal");
+    elec_noise 		=
+      parameters.getParameter< double >("elec_noise");
+    sipm_gain 		=
+      parameters.getParameter< double >("sipm_gain");
+    s_freq 		=
+      parameters.getParameter< double >("qie_sf");
+      
```

Comment:
I've found it very useful to print the set parameters to the logger (in info or even debug mode, is enough) because there is still no (central) protection against a typo in the parameter name you specify in the python config.